src/rdkafka_partition.c

```diff
@@ -2568,7 +2568,21 @@ rd_kafka_topic_partition_list_t *rd_kafka_topic_partition_list_new(int size) {
         return rktparlist;
 }
 
+rd_kafka_topic_partition_t *
+rd_kafka_topic_partition_new_with_topic_id(rd_kafka_Uuid_t topic_id,
+                                           int32_t partition) {
+        rd_kafka_topic_partition_private_t *parpriv;
+        rd_kafka_topic_partition_t *rktpar = rd_calloc(1, sizeof(*rktpar));
+
+        rktpar->partition = partition;
 
+        if (!rd_kafka_Uuid_cmp(topic_id, RD_KAFKA_UUID_ZERO))
+                return rktpar;
```

Comment:
Can we remove this hard coding so that we don't have to remember doing this everytime later? Creating parpriv always when topic_id is needed even with UUID_ZERO?